src/components/molecule_3d.jsx

```diff
@@ -156,45 +172,48 @@ class Molecule3d extends React.Component {
         });
       }
 
-      glviewer.setStyle({ serial: atom.serial }, libStyle);
-    });
+      const libStyleString = JSON.stringify(libStyle);
+      stylesByAtom.set(atom.serial, libStyleString);
 
-    // Shapes
-    this.props.shapes.forEach((shape) => {
-      if (shape.type) {
-        glviewer[`add${shape.type}`](libUtils.getShapeSpec(shape, this.setSelectionTrait));
+      // If the style string for this atom is the same as last time, then no
+      // need to set it again
+      if (this.lastStylesByAtom &&
+        this.lastStylesByAtom.get(atom.serial) === libStyleString) {
```

Comment:
Yeah it's just strings.  I thought I was going to be able to use the `libStyle` object itself as a key, but I need to find that key by value and not by reference, so I had to use the stringified version as the key instead.

I'm still confused about the specifics of the use cases for the new JS data structures.  I think I'm going to write my blog post on it.  Anyway I switched out the maps for objects and the performance seems to be the same but the syntax is a bit nicer, so I'll push that change.